packages/devtools_app/lib/src/common_widgets.dart

```diff
@@ -988,3 +989,20 @@ class _BreadcrumbPainter extends CustomPainter {
         colorScheme != oldDelegate.colorScheme;
   }
 }
+
+class FormattedJson extends StatelessWidget {
+  const FormattedJson({@required this.json});
+
+  static const encoder = JsonEncoder.withIndent('  ');
+
+  final Map<String, dynamic> json;
+
+  @override
+  Widget build(BuildContext context) {
+    final formattedArgs = encoder.convert(json);
+    return Text(
+      formattedArgs.replaceAll('"', ''),
```

Comment:
I would leave in the quotes. Otherwise there is a risk that json like
"Then they said \"hello\"."
will like 
"Then they said \hello\." which is a bit confusing.
You could add a TODO to use a prettier format like YAML. Even though JSON is a bit ugly it is nice in that you can copy and paste it into Dart and it is generally a valid literal.